src/PVRIptvData.cpp

```diff
@@ -468,10 +475,11 @@ bool PVRIptvData::LoadPlayList(void)
 
       iChannelNum++;
 
-      if (iCurrentGroupId > 0)
+      std::vector<int>::iterator it;
+      for (it = iCurrentGroupId.begin(); it != iCurrentGroupId.end(); ++it)
       {
-        channel.bRadio = m_groups.at(iCurrentGroupId - 1).bRadio;
-        m_groups.at(iCurrentGroupId - 1).members.push_back(iChannelIndex);
+        channel.bRadio = m_groups.at(*it - 1).bRadio;
```

Comment:
This is fine.
The lowest possible value for an int in `iCurrentGroupId` is `1` because:

`int iUniqueGroupId    = 0;`
`group.iGroupId = ++iUniqueGroupId;`
`iCurrentGroupId.push_back(iUniqueGroupId);`